explainer.md

```diff
@@ -0,0 +1,147 @@
+APIs and work together. 
+
+The protocol specifically targets the [requirements of second screen Web
+APIs](requirements.md): the [Remote Playback
+API](https://w3c.github.io/remote-playback/) and the [Presentation
+API](https://w3c.github.io/presentation-api/).
+
+A design goal is to re-use existing network and data standards like mDNS, QUIC
+and CBOR as building blocks, versus inventing our own.
+
+The protocol also provides [security and privacy
+guarantees](requirements.md#privacy-and-security), especially with regards to
+other devices on the network.
+
+We assume that the protocol should work on devices with [low-end CPUs and
+constrained memory](device_specs.md), which are typical in special purpose
+devices like smart TVs and connected speakers.  We also assume the protocol will
+be used on [battery-powered devices](requirements.md#hardware-and-efficiency). 
+
+## Non-Goals
+
+A non-goal is to enable every Web application to be able to successfully render
+content on every Open Screen Protocol device.  Devices have a wide range of
+rendering capabilities and policies about what content they wish to support.
+The protocol allows devices to indicate whether they are compatible with a
+specific URL or media type.
+
+## Alternatives Considered
+
+We considered adopting an existing protocol like [Google
+Cast](https://developers.google.com/cast/) or [HbbTV
+2.0](https://www.hbbtv.org/news-events/hbbtv-2-0-specification-released/)
+instead of inventing a new protocol from scratch.  However these had limitations
+that made them less than ideal:
+
+  * They were not free to implement by other vendors.
+  * They did not fully support the Web APIs as written.
+  * They did not have a security model that met the requirements of the Web APIs.
+  * They would be harder to extend to anticipated future use cases,
+    like cross-LAN support or real-time streaming.
+
+## Sample Open Screen Protocol session
+
+As the protocol is not directly exposed to the Web application, we create a
+sample session of CBOR messages that two devices might exchange in response to
+use of the Web APIs.  (Complete sample code for the Web APIs themselves can
+be found in the respective specifications.)
+
+Web application:
+
+```
+const p = new PresentationRequest("https://www.example.com");
+let availability;
+p.getAvailability().then(a => availability = a);
+```
+
+At this point the user agent has received a request to search for devices
+capable of displaying `https://www.example.com`.  It uses mDNS to search for
+Open Screen Protocol endpoints on the LAN and initiates a QUIC connection if it finds one.
+
+If this is the first time connecting to the endpoint, it exchanges
+authentication messages with it.  The user is required to type a password to
+authenticate that the endpoint is the intended target.
+
+
+Open Screen Messages:
+
+```
+TODO
+```
+
+Web application:
+
+```
+let connection;
+p.start().then(c => connection = c);
+c.send("Hello there!");
```

Comment:
Shouldn't this be `connection.send("Hello there!");`?

Reply:
This section was removed.